Extensions/MUC-Sub/XMPPMUCSub.m

```diff
@@ -0,0 +1,699 @@
+//
+//  XMPPMUCSub.m
+//  XMPPFramework
+//
+//  Created by Robert Lohr on 09.10.2017.
+//
+
+#import <Foundation/Foundation.h>
+#import "XMPPMUCSub.h"
+#import "XMPP.h"
+#import "XMPPIDTracker.h"
+#import "XMPPLogging.h"
+#import "XMPPFramework.h"
+#import "XMPPRoom.h"
+
+#if ! __has_feature(objc_arc)
+    #warning This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#if DEBUG
+    static const int xmppLogLevel = XMPP_LOG_FLAG_TRACE; // | XMPP_LOG_FLAG_TRACE;
+#else
+    static const int xmppLogLevel = XMPP_LOG_LEVEL_WARN;
+#endif
+
+static NSString *const XMPPPubSubNamespace = @"http://jabber.org/protocol/pubsub#event";
+static NSString *const XMPPMUCSubNamespace = @"urn:xmpp:mucsub:0";
+static NSString *const XMPPMUCSubFeaturesPrefix = @"urn:xmpp:mucsub:nodes:";
+
+@interface XMPPMUCSub (PrivateAPI)
+
+@end
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma mark -
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+@implementation XMPPMUCSub
+
+- (id)init
+{
+    return [self initWithDispatchQueue:nil];
```

Comment:
XMPPFramework uses `NULL`. `nil` seems to be used for Objective-C objects; is `dispatch_queue_t` an Objective-C or C class?

If I understand [this](https://developer.apple.com/documentation/dispatch/dispatch_queue_t) correctly, it's the former and `nil` is correct.

Reply:
It's a type alias to an Objective-C class, so `nil` is correct.